scripts/Diffraction/isis_powder/pearl.py

```diff
@@ -1,557 +1,140 @@
 from __future__ import (absolute_import, division, print_function)
 
-import os
-
 import mantid.simpleapi as mantid
-import numpy as numpy
-from mantid import config
 
-import isis_powder.routines.common as common
+from isis_powder.routines import common, InstrumentSettings, yaml_parser
+from isis_powder.routines.common_enums import InputBatchingEnum
 from isis_powder.abstract_inst import AbstractInst
-from isis_powder.pearl_routines import fmode_output, pearl_calib_factory, pearl_cycle_factory
-from isis_powder.routines.RunDetails import RunDetails
+from isis_powder.pearl_routines import pearl_algs, pearl_output, pearl_advanced_config, pearl_param_mapping
 
 
 class Pearl(AbstractInst):
+    def __init__(self, **kwargs):
+        basic_config_dict = yaml_parser.open_yaml_file_as_dictionary(kwargs.get("config_file", None))
```

Comment:
So as all kwargs get forwarded to the `InstrumentSettings` class at construction it would subsequently warn (or throw is they choose that) the key was misspelt. 
Additionally the config_file is designed to move several keys out of the scripting window as they were becoming noise. If this key was misspelt those keys would be missing so it would fail too instead of trying to guess.